src/full/Agda/TypeChecking/Errors.hs

```diff
@@ -1248,6 +1249,35 @@ instance PrettyTCM TypeError where
     MultiplePolarityPragmas xs -> fsep $
       pwords "Multiple polarity pragmas for" ++ map pretty (List1.toList xs)
 
+    ConstructorNameOfNonRecord res -> case res of
+      UnknownName -> __IMPOSSIBLE__ -- Turned into NotInScope when the name is resolved
+      name ->
+        let
+          qn :: m Doc
+          (qn, whatis) = case name of
+            DefinedName _ nm _ -> (prettyTCM nm,) case anameKind nm of
+              RecName                  -> __IMPOSSIBLE__
+
+              ConName                  -> "a constructor"
+              CoConName                -> "a constructor"
+              FldName                  -> "a record field"
+              PatternSynName           -> "a pattern synonym"
+              GeneralizeName           -> "a generalized variable"
+              DisallowedGeneralizeName -> "a generalized variable"
+              MacroName                -> "a macro"
+              QuotableName             -> "a quotable name"
+
+              DataName                 -> "a data type"
+              FunName                  -> "a function"
+              AxiomName                -> "a postulate"
+              PrimName                 -> "a primitive"
+              OtherDefName             -> "a defined symbol"
+            VarName a _ -> (prettyTCM a, "a local variable")
+            FieldName (a :| _) -> (prettyTCM a, "the name of an ambiguous projection")
+            ConstructorName _ (a :| _) -> (prettyTCM a, "the name of an ambiguous constructor")
+            PatternSynResName (a :| _) -> (prettyTCM a, "the name of an ambiguous pattern synonym")
```

Comment:
I think "ambiguous" should be dropped from this error, as it does not contribute really to understanding the error.